.github/workflows/build.yml

```diff
@@ -177,13 +186,29 @@ jobs:
       # our end, but PyPI only needs the tarball.
       - name: Upload sdist and wheels to PyPI
         run: |
-          python -m pip install "twine"
-          python -m twine upload --verbose wheels/*.whl sdist/*.tar.gz
+          # The confirmation is tested explicitly in `deploy_test`, so we know
+          # it is either a missing confirmation (so we shouldn't run this job),
+          # 'test' or a valid confirmation.  We don't need to retest the value
+          #  of the confirmation, beyond checking that one existed.
+          if [ ${{ github.event.inputs.confirm_ref }} == 'test' ]; then
+            export TWINE_REPOSITORY=testpypi
+            export TWINE_PASSWORD=${{ secrets.TESTPYPI_TOKEN }}
+          elif [ ${{ github.event.inputs.confirm_ref }} != '' ]; then
+            export TWINE_REPOSITORY=pypi
+            export TWINE_PASSWORD=${{ secrets.PYPI_TOKEN }}
+          else
+            # Exit without deploying
+            echo "Don't update wheels"
+            exit 0
+          fi
+          echo "Deploy to $TWINE_REPOSITORY"
+          echo "python -m pip install "twine""
+          echo "python -m twine upload --verbose wheels/*.whl sdist/*.tar.gz"
```

Comment:
The pipeline seems to still be faling. based on the error message and the previous check I think we need to change `sdist` to `wheels`